scim/resource_group_member.go

```diff
@@ -3,21 +3,127 @@ package scim
 import (
 	"context"
 	"fmt"
+	"strings"
+	"sync"
 
 	"github.com/databricks/databricks-sdk-go/apierr"
 	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 )
 
+type GroupMembersInfo struct {
+	initialized bool
+	members     map[string]struct{}
+	lock        sync.Mutex
+}
+
+type GroupCache struct {
+	cache map[string]*GroupMembersInfo
+	lock  sync.Mutex
+}
+
+func NewGroupsCache() *GroupCache {
+	return &GroupCache{
+		cache: make(map[string]*GroupMembersInfo),
+		lock:  sync.Mutex{},
+	}
+}
+
+func (gc *GroupCache) getOrCreateGroupInfo(groupID string) *GroupMembersInfo {
+	gc.lock.Lock()
+	defer gc.lock.Unlock()
+
+	groupInfo, exists := gc.cache[groupID]
+	if !exists {
+		groupInfo = &GroupMembersInfo{
+			initialized: false,
+			members:     make(map[string]struct{}),
+			lock:        sync.Mutex{},
+		}
+		gc.cache[groupID] = groupInfo
+	}
+	return groupInfo
+}
+
+func (gc *GroupCache) GetMembers(api GroupsAPI, groupID string) (map[string]struct{}, error) {
```

Comment:
same, does this need to be exported?
```suggestion
func (gc *GroupCache) getMembers(api GroupsAPI, groupID string) (map[string]struct{}, error) {
```